packages/utils/cli/bin/spark-generate.mjs

```diff
@@ -1,56 +1,156 @@
 
-if (!component) {
-  showError('Argument "component" must be defined!')
+  return template
 }
-if (!WORDS_ONLY_REGEX.test(component)) {
-  showError('"component" name must contain letters and dash symbols only')
+
+async function promptPackageDescription() {
+  const description = await prompt.text({
+    message: 'Describe your package (short description):',
+    initialValue: '',
+    validate(value) {
+      if (!value) return `You package must have a description`
+      if (value.length < 10) return ERRORS.INVALID_DESCRIPTION
+    },
+  })
+
+  if (prompt.isCancel(description)) showError(ERRORS.ABORT)
+
+  return description
 }
 
-const componentDir = `/packages/components/${component}/`
-const componentPath = `${BASE_DIR}${componentDir}`
-const templatesPattern = fileURLToPath(new URL('../templates/**/*.js', import.meta.url))
+/**
+ * Program starts here
+ */
+prompt.intro(`Generate Spark package`)
 
-glob(templatesPattern, async (err, res) => {
-  if (err) showError(err)
-  if (res) {
-    const templateContents = res.map(templatePath =>
-      import(templatePath).then(module => ({
-        path: templatePath
-          .replace(/(.*)\/templates\//, componentPath)
-          .replace('Component', pascalCase(component))
-          .replaceAll(/\[|\]|\.js$/g, ''),
-        content: module.default({ component, description }),
-      }))
-    )
+const name = await promptPackageName()
+const template = await promptPackageTemplate()
+const description = await promptPackageDescription()
 
-    const filesToWrite = await Promise.all(templateContents)
+const packagePath = packageUtils.getFullPath(name, template)
 
-    Promise.all(filesToWrite.map(writeFile)).then(() => {
-      log('All component files has been properly written!')
-    })
-  }
-})
+switch (template) {
+  case TEMPLATE_TYPE.COMPONENT:
+    generateComponentPackage(name, description)
+    break
+  case TEMPLATE_TYPE.HOOK:
+    generateHookPackage(name, description)
+    break
+}
+
+prompt.outro(`Generating package...`)
+
+function generateComponentPackage(name, description) {
+  const templatesPattern = fileURLToPath(new URL('../templates/**/*.js', import.meta.url))
+
+  glob(templatesPattern, async (err, res) => {
+    if (err) showError(err)
+    if (res) {
+      const templateContents = res.map(templatePath =>
+        import(templatePath).then(module => ({
+          path: templatePath
+            .replace(/(.*)\/templates\//, packagePath)
+            .replace('Component', pascalCase(name))
+            .replaceAll(/\[|\]|\.js$/g, ''),
+          content: module.default({
+            component: name,
+            description: description,
+          }),
+        }))
+      )
+
+      const filesToWrite = await Promise.all(templateContents)
+
+      Promise.all(filesToWrite.map(writeFile)).then(() => {
+        log.success('All package files has been properly written!')
+      })
+    }
+  })
+}
+
+function generateHookPackage(name, description) {
```

Comment:
If this file starts to grow a lot, we could split it moving these parts away from here.

Reply:
I was thinking about a `generate` folder with all files, so that it's clearcly seprate from other commands in the future.